pkg/storage/etcd/etcd_watcher.go

```diff
@@ -359,9 +359,6 @@ func (w *etcdWatcher) sendAdd(res *etcd.Response) {
 		return
 	}
 	action := watch.Added
-	if res.Node.ModifiedIndex != res.Node.CreatedIndex {
```

Comment:
That's why we should document it. See #36797
Nonetheless, I have added comments in sendResult() as well.